refactor/data/app/api/api_v0/endpoints/workflow.py

```diff
@@ -29,78 +29,73 @@
     ResultPickle,
     UpdateResultResponse,
 )
-from fastapi import APIRouter
+from fastapi import APIRouter, UploadFile, HTTPException
+from fastapi.responses import FileResponse, StreamingResponse
 
 router = APIRouter()
 
-# For now we don't return this, instead just a bytes-like pickle object
-mock_result_object_instance = {
-    "dispatch_id": "",
-    "results_dir": "",
-    "status": "COMPLETED",
-    "graph": {
-        "links": [{"edge_name": "data", "param_type": "kwarg", "source": 0, "target": 1}],
-        "nodes": [
-            {
-                "name": "load_data",
-                "metadata": {"executor": "<LocalExecutor>"},
-                "function_string": "@ct.electron\ndef load_data():\n    iris = datasets.load_iris()\n    perm = permutation(iris.target.size)\n    iris.data = iris.data[perm]\n    iris.target = iris.target[perm]\n    return iris.data, iris.target\n\n\n",
-                "start_time": "2022-02-28T23:21:17.844268+00:00",
-                "end_time": "2022-02-28T23:21:17.853741+00:00",
-                "status": "COMPLETED",
-                "output": ["[[5.  3.4 1.5 0.2]]"],
-                "error": None,
-                "sublattice_result": None,
-                "stdout": "",
-                "stderr": "",
-                "id": 0,
-                "doc": None,
-                "kwargs": None,
-            }
-        ],
-    },
-}
-
 
 @router.get(
     "/results/{dispatch_id}",
     status_code=200,
-    response_model=ResultPickle,
+    response_class=FileResponse,
     responses={
-        404: {"model": HTTPExceptionSchema, "description": "The result was not found"},
-    },
+        404: {"model": HTTPExceptionSchema, "description": "Result was not found"},
+        200: {
+            "content": {"application/octet-stream": {}},
+            "description": "Return binary content of file.",
+        }
+    }
 )
 def get_result(
     *,
     dispatch_id: str,
 ) -> Any:
     """
-    Get a result object
+    Get a result object from the database
```

Comment:
Technically the results objects are not coming from the database, they are coming from the object store. But maybe that's a detail that doesn't need to be here.

Reply:
Gotcha, will update 👍